robot/basestation/static/js/elements/navigation/nav_funs.js

```diff
@@ -0,0 +1,57 @@
+/*
+This is a piece of code created by @Waleed-HM which I converted to JS
+*/
+
+
+function direction_to_rover(p1_lat, p1_lon, p2_lat, p2_lon) {
+    /*
+    This is a function to calculate the direction from p1 (point1) to p2 (point2)
+    It uses the Haverside formulas
+    Inputs are p1 and p2 latitudes and longitudes in degrees
+    Output is the direction from p1 to p2 in degrees ranging from 0 to 360
+    Measured from the N-S line, left-handed rule is used (E=90, W=270)
+    */
+
+    p1_lat = p1_lat * Math.PI / 180
+    p1_lon = p1_lon * Math.PI / 180
+    p2_lat = p2_lat * Math.PI / 180
+    p2_lon = p2_lon * Math.PI / 180
+
+    let delta_lat = p2_lat - p1_lat
+    let delta_lon = p2_lon - p1_lon
+
+    let X = Math.cos(p2_lat) * Math.sin(delta_lon)
+    let Y = Math.cos(p1_lat) * Math.sin(p2_lat) - Math.sin(p1_lat) * Math.cos(p2_lat) * Math.cos(delta_lon)
+
+    let Real_dir = Math.atan2(X, Y)
```

Comment:
variable names should be in camelCase. Only time it should start with a Capital letter is when it's the name of a class like Math